src/screens/MapPage/MapPage.js

```diff
@@ -23,6 +28,27 @@ export default function MapPage() {
           <MapDetail
             coordinates={miniMapCoords}
             onClose={() => setCoords(null)}
+            setActiveSubject={setActiveSubject}
+            setShowSubjectsModal={setShowSubjectsModal}
+          />
+        </Layer>
+      )}
+
+      {activeSubject && (
+        <Layer>
```

Comment:
I would define onEsc for the Layer:
```suggestion
        <Layer
          onEsc={() => setActiveSubject(null)}
        >
```